chore(wallet)_: unit tests for SendTransaction, transaction manager API cleanup

IvanBelyakoff opened the pull request:
Improvements in code structure to make package more clean and testable:
- exported API methods left at the same place
- private methods moved to helpers.go
- stuff for testing moved to testutils.go
- created storage interface with clean API and multi transaction related db calls moved to MultiTransactionDBStorage implementation
- created dummy in-mem storage for tests with multi transactions

Test related changes:
- written unit tests for MultiTransactionDBStorage
- mocked Bridge interface
- mocked Transactor interface
- written unit tests for `TransactionManager.SendTransactions` method

Closes #5148

IvanBelyakoff committed:
chore(wallet)_: code structure improved for multi_transaction manager

- exported API methods left at the same place
- private methods moved to helpers.go
- stuff for testing moved to testutils.go
- created storage interface with clean API and multi transaction related db calls moved
  to MultiTransactionDBStorage implementation
- created dummy in-mem storage for tests with multi transactions
- written tests for MultiTransactionDBStorage

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (42)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 105bb581 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/1/) | 2024-05-23 19:37:30 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 105bb581c2c9c785679d794a092e5777b96319a7 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/1/) | 2024-05-23 19:39:11 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240523-193450-105bb581c2c9c785679d794a092e5777b96319a7-pr5221.zip) |
| :heavy_check_mark: | 105bb581c2c9c785679d794a092e5777b96319a7 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/1/) | 2024-05-23 19:40:13 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240523-193447-105bb581c2c9c785679d794a092e5777b96319a7-pr5221.zip) |
| :heavy_check_mark: | 105bb581c2c9c785679d794a092e5777b96319a7 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/1/) | 2024-05-23 19:41:06 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240523-193446-105bb581c2c9c785679d794a092e5777b96319a7-pr5221.aar) |
| | | | | | | |
| :heavy_check_mark: | 766084b836fee89ea9a086987e85aad2612ea0a9 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/2/) | 2024-05-23 19:51:52 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240523-194935-766084b836fee89ea9a086987e85aad2612ea0a9-pr5221.aar) |
| :heavy_check_mark: | 766084b836fee89ea9a086987e85aad2612ea0a9 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/2/) | 2024-05-23 19:53:06 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240523-194935-766084b836fee89ea9a086987e85aad2612ea0a9-pr5221.zip) |
| :heavy_check_mark: | 766084b836fee89ea9a086987e85aad2612ea0a9 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/2/) | 2024-05-23 19:53:57 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240523-194939-766084b836fee89ea9a086987e85aad2612ea0a9-pr5221.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 766084b8 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/2/) | 2024-05-23 19:52:38 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 62d211448b9bca6fbc88319e777b8847810a1c5d | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/3/) | 2024-05-23 21:38:13 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240523-213616-62d211448b9bca6fbc88319e777b8847810a1c5d-pr5221.aar) |
| :heavy_check_mark: | 62d211448b9bca6fbc88319e777b8847810a1c5d | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/3/) | 2024-05-23 21:38:31 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240523-213616-62d211448b9bca6fbc88319e777b8847810a1c5d-pr5221.zip) |
| :heavy_check_mark: | 62d211448b9bca6fbc88319e777b8847810a1c5d | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/3/) | 2024-05-23 21:39:31 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240523-213616-62d211448b9bca6fbc88319e777b8847810a1c5d-pr5221.zip) |
| | | | | | | |
| :heavy_check_mark: | 62d21144 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/3/) | 2024-05-23 22:20:39 | ~44 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/3//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 4f834cb6 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/4/) | 2024-05-26 15:03:04 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4f834cb6bcdee25dad2128f62556a974eef85739 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/4/) | 2024-05-26 15:04:28 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240526-150151-4f834cb6bcdee25dad2128f62556a974eef85739-pr5221.zip) |
| :heavy_check_mark: | 4f834cb6bcdee25dad2128f62556a974eef85739 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/4/) | 2024-05-26 15:04:37 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240526-150146-4f834cb6bcdee25dad2128f62556a974eef85739-pr5221.aar) |
| :heavy_check_mark: | 4f834cb6bcdee25dad2128f62556a974eef85739 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/4/) | 2024-05-26 15:06:16 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240526-150151-4f834cb6bcdee25dad2128f62556a974eef85739-pr5221.zip) |
| | | | | | | |
| :heavy_multiplication_x: | c86f59a3 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/5/) | 2024-05-26 15:13:48 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c86f59a38ecfc102b1a870ca95f7fdcbc649df82 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/5/) | 2024-05-26 15:14:56 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240526-151221-c86f59a38ecfc102b1a870ca95f7fdcbc649df82-pr5221.aar) |
| :heavy_check_mark: | c86f59a38ecfc102b1a870ca95f7fdcbc649df82 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/5/) | 2024-05-26 15:14:58 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240526-151225-c86f59a38ecfc102b1a870ca95f7fdcbc649df82-pr5221.zip) |
| :heavy_check_mark: | c86f59a38ecfc102b1a870ca95f7fdcbc649df82 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/5/) | 2024-05-26 15:15:51 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240526-151221-c86f59a38ecfc102b1a870ca95f7fdcbc649df82-pr5221.zip) |
| | | | | | | |
| :heavy_multiplication_x: | e1ce3ca6 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/6/) | 2024-05-26 15:36:50 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e1ce3ca621b246460d1af0a9b74151bdc27275f1 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/6/) | 2024-05-26 15:37:35 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240526-153521-e1ce3ca621b246460d1af0a9b74151bdc27275f1-pr5221.aar) |
| :heavy_check_mark: | e1ce3ca621b246460d1af0a9b74151bdc27275f1 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/6/) | 2024-05-26 15:38:01 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240526-153526-e1ce3ca621b246460d1af0a9b74151bdc27275f1-pr5221.zip) |
| :heavy_check_mark: | e1ce3ca621b246460d1af0a9b74151bdc27275f1 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/6/) | 2024-05-26 15:38:58 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240526-153526-e1ce3ca621b246460d1af0a9b74151bdc27275f1-pr5221.zip) |
| | | | | | | |
| :heavy_check_mark: | 3a91b00d46155c3db923b45e625388f52b402b1b | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/7/) | 2024-05-27 04:52:59 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240527-045101-3a91b00d46155c3db923b45e625388f52b402b1b-pr5221.aar) |
| :heavy_check_mark: | 3a91b00d46155c3db923b45e625388f52b402b1b | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/7/) | 2024-05-27 04:53:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240527-045105-3a91b00d46155c3db923b45e625388f52b402b1b-pr5221.zip) |
| :heavy_check_mark: | 3a91b00d46155c3db923b45e625388f52b402b1b | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/7/) | 2024-05-27 04:54:21 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240527-045101-3a91b00d46155c3db923b45e625388f52b402b1b-pr5221.zip) |
| | | | | | | |
| :heavy_check_mark: | 3a91b00d | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/7/) | 2024-05-27 05:33:53 | ~42 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b3a39099c20028240ab81d542e66c2ba949f85fe | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/8/) | 2024-05-27 15:32:11 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240527-152934-b3a39099c20028240ab81d542e66c2ba949f85fe-pr5221.zip) |
| :heavy_check_mark: | b3a39099c20028240ab81d542e66c2ba949f85fe | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/8/) | 2024-05-27 15:32:26 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240527-152934-b3a39099c20028240ab81d542e66c2ba949f85fe-pr5221.aar) |
| :heavy_check_mark: | b3a39099c20028240ab81d542e66c2ba949f85fe | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/8/) | 2024-05-27 15:33:47 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240527-152934-b3a39099c20028240ab81d542e66c2ba949f85fe-pr5221.zip) |
| | | | | | | |
| :heavy_check_mark: | b3a39099 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/8/) | 2024-05-27 16:12:38 | ~42 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2a5f67d3c94e77f95854c8a31d6f5d2a45808835 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/9/) | 2024-05-31 05:34:43 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240531-053201-2a5f67d3c94e77f95854c8a31d6f5d2a45808835-pr5221.zip) |
| :heavy_check_mark: | 2a5f67d3c94e77f95854c8a31d6f5d2a45808835 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/9/) | 2024-05-31 05:34:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240531-053201-2a5f67d3c94e77f95854c8a31d6f5d2a45808835-pr5221.aar) |
| :heavy_check_mark: | 2a5f67d3c94e77f95854c8a31d6f5d2a45808835 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/9/) | 2024-05-31 05:36:21 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240531-053201-2a5f67d3c94e77f95854c8a31d6f5d2a45808835-pr5221.zip) |
| | | | | | | |
| :heavy_check_mark: | 2a5f67d3 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/9/) | 2024-05-31 06:15:17 | ~43 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | eec24a5bf65c0eec5b4ab2c50d58054fa0f3a805 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/10/) | 2024-05-31 06:34:03 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240531-062939-eec24a5bf65c0eec5b4ab2c50d58054fa0f3a805-pr5221.zip) |
| :heavy_check_mark: | eec24a5bf65c0eec5b4ab2c50d58054fa0f3a805 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/10/) | 2024-05-31 06:35:42 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240531-062939-eec24a5bf65c0eec5b4ab2c50d58054fa0f3a805-pr5221.zip) |
| :heavy_check_mark: | eec24a5bf65c0eec5b4ab2c50d58054fa0f3a805 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/10/) | 2024-05-31 06:35:54 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240531-062939-eec24a5bf65c0eec5b4ab2c50d58054fa0f3a805-pr5221.aar) |
| | | | | | | |
| :heavy_check_mark: | 5b553dd3f27508c0d4a0e2919c9e77c766203ad9 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5221/11/) | 2024-05-31 06:53:48 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240531-065020-5b553dd3f27508c0d4a0e2919c9e77c766203ad9-pr5221.zip) |
| :heavy_check_mark: | 5b553dd3f27508c0d4a0e2919c9e77c766203ad9 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5221/11/) | 2024-05-31 06:53:49 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240531-065020-5b553dd3f27508c0d4a0e2919c9e77c766203ad9-pr5221.zip) |
| :heavy_check_mark: | 5b553dd3f27508c0d4a0e2919c9e77c766203ad9 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5221/11/) | 2024-05-31 06:55:58 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240531-065020-5b553dd3f27508c0d4a0e2919c9e77c766203ad9-pr5221.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | eec24a5b | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/10/) | 2024-05-31 07:11:57 | ~42 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5b553dd3 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/11/) | 2024-05-31 07:53:40 | ~41 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5221/11//consoleText) |

IvanBelyakoff committed:
test(wallet)_: created Transactor interface

- Moved some methods from Transactor to users of it to clean interface.
- Mocked Bridge interface and Transactor interface for tests
- Wrote unit tests for SendTransaction

saledjenic commented on services/wallet/transfer/helpers.go:
```diff
@@ -0,0 +1,245 @@
+	return multiTransactions, nil
+}
+
+func (tm *TransactionManager) ensureTransactionsSigned(signatures map[string]SignatureDetails) error {
```
@IvanBelyakoff could you please rename this function to `addSignaturesToTransacations` or `associateSignaturesToTransactions`, cause this function doesn't sign transactions?

saledjenic commented on services/wallet/transfer/helpers.go:
```diff
@@ -0,0 +1,245 @@
+	return multiTransaction
+}
+
+func (tm *TransactionManager) buildTransactions(bridges map[string]bridge.Bridge) ([]string, error) {
```
@IvanBelyakoff I personally wouldn't move functions that refer to certain types like `func (tm *TransactionManager) ...` to the file named "helpers". Please check if there is a chance to pull them out from the `TransactionManager` and keep them here or move them back.

saledjenic commented on services/wallet/transfer/helpers.go:
```diff
@@ -0,0 +1,245 @@
+	}
+}
+
+func (tm *TransactionManager) sendTransactions(
```
For this one we're good to remove `(tm *TransactionManager)` and keep it here.

saledjenic reviewed (approved):
@IvanBelyakoff nice separation and a good step forward towards testing transactions part.
